File: Scripts/laptop/function-composition/python/function-composition.py
from math import sin, asin

def executeTask(i):
    compose = lambda f, g: lambda x: f( g(x) )
    sin_asin = compose(sin, asin)
    sin_asin(0.5)
    return i + 1

r = 1
for i in xrange(1000000000):
	r = executeTask(i + r)

File: Scripts/laptop/function-composition/perl/function-composition.pl
sub compose {
    my ($f, $g) = @_;

    sub {
        $f -> ($g -> (@_))
    };
}

use Math::Trig;

my $r = 1;

sub executeTask {
	my ($i) = @_;
    	compose(sub {sin $_[0]}, \&asin)->(0.5);
	return ($i + 1);
}

for($a = 0; $a < 1000000000; $a += 1) { 
	$r = executeTask($a + $r);
}
